### PR TITLE
[Seq2SeqTrainer] Move import to init to make file self-contained

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Seq2SeqTrainer can be used as an independent file if no `label_smoothing` is done. This PR move the import to the init to make it possible to simple download this file and use it as it is without any dependencies for standard seq2seq training.

Would be awesome if @patil-suraj  and @stas00 could review as well :-)

### CODE REVIEW

**Comment by patil-suraj** [comment]
I'm okay with this, but users will still need to copy paste or rewrite  `Seq2SeqTrainingArguments`   and pass those as `args` rather than default `TrainingArguments`, since we assume `args` is `Seq2SeqTrainingArguments` in multiple methods

**Review by sshleifer — APPROVED** [review]

File: examples/seq2seq/seq2seq_trainer.py [hunk 1]
```diff
@@ -64,6 +58,17 @@ def __init__(self, config=None, data_args=None, *args, **kwargs):
                 f"The `config.pad_token_id` is `None`. Using `config.eos_token_id` = {self.config.eos_token_id} for padding.."
             )
 
+        if self.args.label_smoothing == 0:
+            self.loss_fn = torch.nn.CrossEntropyLoss(ignore_index=self.config.pad_token_id)
+        else:
+            # dynamically import label_smoothed_nll_loss
+            try:
```
Comment (stas00) [inline examples/seq2seq/seq2seq_trainer.py:65]: We moved all seq2seq scripts/tests to not need the try/except workaround, and just `import from utils`, see:
https://github.com/huggingface/transformers/pull/7274/files#diff-36ebee2221416f906bccc099a8958f6814f51a1dc8f2b07a75ffd19bd0d1ddf6
this script looks like a new file that was copied from some old style code.

Of course, your PR didn't introduce it, but just as well let's use the clean new style.

Comment (patrickvonplaten) [reply examples/seq2seq/seq2seq_trainer.py:65]: Thanks for letting me know! -> will open a PR to correct it :-)

Comment (patrickvonplaten) [reply examples/seq2seq/seq2seq_trainer.py:65]: Pinged you on the PR :-) https://github.com/huggingface/transformers/pull/8254

**Pull request merged**